src/components/organisms/NavBarSchedule/NavBarSchedule.tsx

```diff
@@ -105,21 +96,25 @@ export const NavBarSchedule: FC<NavBarScheduleProps> = ({ isVisible }) => {
 
   const getEventLocation = useCallback(
     (locString: string): VenueLocation => {
-      const [venueId, roomTitle = ""] = extractLocation(locString);
-      const venueTitle = relatedVenues.find((venue) => venue.id === venueId)
+      const [venueId, roomTitle] = extractLocation(locString);
+      const venueName = relatedVenues.find((venue) => venue.id === venueId)
         ?.name;
-      return { venueId, roomTitle, venueTitle };
+      return { venueId, roomTitle, venueName };
     },
     [relatedVenues]
   );
 
   const schedule: ScheduleDay = useMemo(() => {
-    const dayStart = addDays(startOfToday(), selectedDayIndex);
+    const startOfSelectedDay = addDays(startOfToday(), selectedDayIndex);
     const daysEvents = relatedVenueEvents
-      .filter(isEventWithinDate(selectedDayIndex === 0 ? Date.now() : dayStart))
-      .map(prepareForSchedule(dayStart, userEventIds));
-
-    const locatedEvents: LocatedEvents[] = Object.entries(
+      .filter(
+        isEqual(startOfToday(), startOfSelectedDay)
+          ? isEventLaterToday
+          : isEventWithinDate(startOfSelectedDay)
+      )
+      .map(prepareForSchedule(startOfSelectedDay, userEventIds));
```

Comment:
Tired 🧠 can't follow/assess this logic right now.

@mike-lvov Does this look fine to you?

Reply:
had to rewrite that piece a bit after https://github.com/sparkletown/sparkle/pull/1419 was merged.
https://github.com/sparkletown/sparkle/pull/1426/commits/3459d45080f5f3882113c773346fa406fe9eddab

Reply:
Rewritten version looks cleaner IMO